### PR TITLE
Hass nabucasa.0.37.1

### PR DESCRIPTION
**Pull request by Scriptkiddi**
###### Motivation for this change
upgrade hass-nabucasa and deps

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/python-jose/default.nix [hunk 1]
```diff
@@ -23,6 +23,11 @@ buildPythonPackage rec {
     cryptography # optional dependency, but needed in tests
   ];
 
+  patches = [
+    # relax ecdsa deps
+    ./ecdsa.patch
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-jose/default.nix:29]:
```suggestion
  patchPhase = ''
    substituteInPlace requirements.txt \
      --replace 'ecdsa<0.15' 'ecdsa' \
      --replace 'ecdsa <0.15' 'ecdsa'
  '';
```

and the patch can be removed. This breaks less often and is better for simple changes like this.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/python-jose/default.nix (hunk 1 above)
Comment (Scriptkiddi) [reply pkgs/development/python-modules/python-jose/default.nix]: thanks :D didnt know this was possible

**Review by mweinelt — CHANGES_REQUESTED** [review]
I see multiple disabled test cases which have had its issues fixed. Please reenable them.

**Comment by Scriptkiddi** [comment]
> I see multiple disabled test cases which have had its issues fixed. Please reenable them.

hey I can only find one, and that one is still needed because the issue is closed but not fixed in that 3.2.0 release

**Comment by mweinelt** [comment]
- The hass-nabucasa test suite runs on python38.
- The `test_key_too_short` issue has been closed and the result was the ecdsa<0.15 pin https://github.com/mpdavis/python-jose/issues/176#issuecomment-666005314, so we might want to override ecdsa.

**Comment by Scriptkiddi** [comment]
the test to short issue has been closed, but the test still requires ecdsa<0.15 which we dont have in nixpkgs since its 0.15

**Comment by mweinelt** [comment]
Yeah, so the idea is that we pin an earlier version, like 0.14.1, locally in the package, to make it work, since the package apparently isn't compatible with newer ecdsa versions.

**Comment by Scriptkiddi** [comment]
asynctest is still broken on python 3.8 as far as i read this issue https://github.com/Martiusweb/asynctest/issues/132
so enabling it would result in failing tests?

**Comment by Scriptkiddi** [comment]
done reenablet tests for python 3.8

**Review by mweinelt — APPROVED** [review]
LGTM.

**Pull request merged**